components/FooterPlayer/FullScreenPlayer/FullScreenPlayer.tsx

```
@@ -0,0 +1,134 @@
+import styles from "./FullScreenPlayer.module.scss";
+import { PlayPauseButton } from "../../PlayPauseButton/PlayPauseButton";
+import { Agenda } from "../../Agenda/Agenda";
+import React, { CSSProperties } from "react";
+import {
+  faAngleDown,
+  faBackward,
+  faBroadcastTower,
+  faForward,
+  faShareAlt,
+} from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { usePlayerState } from "../../../providers/PlayerProvider/usePlayerState";
+import { usePlayerControls } from "../../../providers/PlayerProvider/usePlayerControls";
+import { useLivePlayer } from "../../../hook/useLivePlayer";
+import { usePlayerBindings } from "../AudioPlayer/usePlayerBinding";
+import { Seeker } from "../AudioPlayer/Seeker/Seeker";
+
+interface FullScreenPlayerProps {
+  visible: boolean;
+  onClose: any;
+  onShare: () => void;
+}
+
+export const FullScreenPlayer = React.forwardRef<
+  HTMLAudioElement,
+  FullScreenPlayerProps
+>(function FullScreenPlayer(props, ref) {
+  const wrapperStyle: CSSProperties = {
+    top: props.visible ? "0" : "100%",
+    transitionDelay: props.visible ? "0.3s" : "0s",
+  };
+
+  const { isLive, toggleLive } = useLivePlayer();
+  const { isPlaying, isPaused, isStopped, songTitle, artist, imageUrl } =
+    usePlayerState();
+  const { pause, resume } = usePlayerControls();
+  // @ts-expect-error
```

Should be escaped early if no ref available, instead of ignoring the error

---

What is the error?

---

this is actually a bit more complicated, it's some TS shit... need to research

---

But it works for now right? let's see if this affect us somehow and for now keep it?

---

yeah yeah it works, just need to make a small TS types fix for it to work properly